include/roboteam_utils/Line.h

```diff
@@ -1,101 +1,126 @@
+     * @param p2 An arbitrary other point on the first line (make sure that it is different than p1).
+     * @param pointOnLine A point that is located on the given line.
+     * @return The value t such that p1 + (p2 - p1) * t = pointOnLine
      */
-    [[nodiscard]] double intercept() const;
+    static float relativePosition(const Vector2 p1, const Vector2 p2, const Vector2 pointOnLine);
 
-    /**
-     * @brief Gets the direction of the Line
+    /* 	 ______   _______  _______  ______     _______  _______  ______   _______
+     *	(  __  \ (  ____ \(  ___  )(  __  \   (  ____ \(  ___  )(  __  \ (  ____ \
+     *	| (  \  )| (    \/| (   ) || (  \  )  | (    \/| (   ) || (  \  )| (    \/
+     *	| |   ) || (__    | (___) || |   ) |  | |      | |   | || |   ) || (__
+     *	| |   | ||  __)   |  ___  || |   | |  | |      | |   | || |   | ||  __)
+     *	| |   ) || (      | (   ) || |   ) |  | |      | |   | || |   ) || (
+     *	| (__/  )| (____/\| )   ( || (__/  )  | (____/\| (___) || (__/  )| (____/\
+     *	(______/ (_______/|/     \|(______/   (_______/(_______)(______/ (_______/
      *
-     * @return Vector2 Vector representation of the direction of this vector
+     * The functions below are dead. Remove this tag if you use any of the functions and make sure to remove this tag at other places as well that will become alive by using any of
+     *the function below. Do not read/document/redesign/analyse/test/optimize/etc. any of this code, because it is a waste of your time! This code was not removed or placed at
+     *another branch, because other software developers are very attached to this code and are afraid that this code might be used at some day (but I think it won't be used at all
+     *and should be removed).
      */
-    [[nodiscard]] Vector2 direction() const;
 
     /**
-     * @brief Gets a pair of coefficients
+     * @brief Gets the intersection of the lines
+     * See https://en.wikipedia.org/wiki/Line%E2%80%93line_intersection for help
      *
-     * @return std::pair<double, double> Pair of doubles where .first == slope() and .second == intercept()
+     * @param line LineSegment to get an intersection from
+     * @return std::shared_ptr<Vector2> Vector representation of this intersection
      */
-    [[nodiscard]] std::pair<double, double> coefficients() const;
+    // [[nodiscard]] std::optional<Vector2> intersects(const LineSegment &line) const;
 
     /**
-     * @brief Checks whether line is vertical
+     * @brief Checks whether \ref line intersects `this`
      *
-     * @return true True if line is vertical, will be true if isPoint is true
-     * @return false False if line is not vertical
+     * @param line Line to check against
+     * @return true True if \ref line intersects `this`
+     * @return false False if \ref line does not intersect `this`
      */
-    [[nodiscard]] bool isVertical() const;
+    // [[nodiscard]] bool doesIntersect(const LineSegment &line) const;
```

Comment:
I strongly disagree with commenting these functions, they are member functions that make a lot of sense to have, at least, to me, so I would not consider this dead code. If you want to hate on static utility functions that are unused, that is another matter. But member functions like this are not dead code, as long as the function makes sense to have for this class